Source/src/core/GameObject.cpp

```diff
@@ -571,6 +578,19 @@ Hachiko::GameObject* Hachiko::GameObject::Find(UID id) const
     return nullptr;
 }
 
+Hachiko::Component* Hachiko::GameObject::GetComponent(Component::Type type) const
```

Comment:
Why we need a new GetComponent method?

Reply:
This is part of what solved the header dependency with ComponentImage in scripting